src/Microsoft.Framework.OptionsModel/ConfigurationBinder.cs

```
@@ -0,0 +1,253 @@
+        {
+            configuration = configuration.GetSubKey(property.Name);
+
+            if (property.GetMethod == null || !property.GetMethod.IsPublic)
+            {
+                // We don't support set only properties
+                return;
+            }
+
+            var propertyValue = property.GetValue(propertyOwner);
+            var hasPublicSetter = property.SetMethod != null && property.SetMethod.IsPublic;
+
+            if (propertyValue == null && !hasPublicSetter)
+            {
+                // Property doesn't have a value and we cannot set it so there is no
+                // point in going further down the graph
+                return;
+            }
+
+            propertyValue = BindType(property.PropertyType, propertyValue, configuration);
+
+            if (propertyValue != null && hasPublicSetter)
+            {
+                property.SetValue(propertyOwner, propertyValue);
+            }
+        }
+
+
+        private static object BindType(Type type, object typeInstance, IConfiguration configuration)
+        {
+            var configValue = configuration.Get(null);
+            var typeInfo = type.GetTypeInfo();
+
+            if (configValue != null)
+            {
+                // Leaf nodes are always reinitialized
+                return CreateValueFromConfiguration(type, configValue, configuration);
+            }
+            else
+            {
+                var subkeys = configuration.GetSubKeys();
+                if (subkeys.Count() != 0)
+                {
+                    if (typeInstance == null)
+                    {
+                        try
+                        {
+                            typeInstance = Activator.CreateInstance(type);
+                        }
+                        catch
+                        {
+                            // Cannot create the value so we cannot populate it
```

There is already a bug on that (you logged it): https://github.com/aspnet/Options/issues/55

I didn't want to change the current functionality which ignores errors. I think we can address the empty catches as part of the other bug